docs/concepts/state.md

```diff
@@ -150,6 +150,47 @@ feedZebra(ctx: StateContext<ZooStateModel>, action: FeedZebra) {
 }
 ```
 
+The `patchState` function can also be called with a patching function which will be given the
+existing state and should return the new state. 
+All immutability concerns need to be honoured by this function. 
+For example you could use a library like [immer](https://github.com/mweststrate/immer) that can 
+handle these concerns for you and provide a different way of expressing your immutable update 
+through direct mutation of a draft object. Here is the example from above expressed in this way:
+```TS
```

Comment:
Add simple example, so that people understand why the operator's patch function appeared

Before

```ts
@Action(MyAction)
public addValue(ctx: StateContext, { payload }: MyAction) {
  ctx.setState({ ...ctx.getState(), value: payload  });
}
```

After

```ts
@Action(MyAction)
public addValue(ctx: StateContext, { payload }: MyAction) {
  ctx.patchState((state) => ({ ...state, value: payload }));
}
```

Reply:
Great idea!

Reply:
done!